### PR TITLE
Netsim multiget2

### PR DESCRIPTION
**Pull request by philbogle**
Add a Sinatra API for fetching of multiple tables in a single request and also specifying the starting row id for each table.

A typical GET request URL might look like:  `/v3/netsim/myshard?t[]=table1&t[]=table2@2&t[]=table3@2`

The underlying functionality is provided by `RedisTable.get_all_tables`

```
RedisTable.get_tables(redis, 'shard1', {'table' => 1, 'table2' => 1})
==> {'table' =>
          {'rows' => [{'name' => 'alice', 'male' => false, 'id' => 1}, {'bar' => 3, 'id' => 3}]},
        'table2' =>
          {'rows' => [{'name' => 'bob',  'male' => true, 'id' => 1}]}}
```

### CODE REVIEW

**Comment by philbogle** [comment]
Brad please take a look when you are free (but no hurry on this)

File: shared/middleware/helpers/redis_table.rb [hunk 1]
```diff
@@ -66,6 +66,46 @@ def to_a
...
+  # Note that the result map will always include a row for a requested table even
+  # if none of the rows passed the min_id test.
+  #
+  # For example:
+  #
+  # RedisTable.get_tables(redis, "shard", {'t1' => 1, 't2' => 3})
+  # =>  {'t1' => {'rows' => [{..}, <etc> }
+
+  # @param [Redis] redis The redis client.
+  # @param []
```
Comment (islemaster) [inline shared/middleware/helpers/redis_table.rb]: `@param [String] shard_id` I assume.

**Comment by islemaster** [comment]
I like it :+1:

**Comment by philbogle** [comment]
Brad, PTAL. I went ahead and added the code for Sinatra REST API to this PR since it was smaller and simpler than I anticipated.

File: shared/middleware/net_sim_api.rb [hunk 2]
```diff
@@ -73,6 +74,18 @@ def get_table(shard_id, table_name)
     get_table(shard_id, table_name).fetch(id.to_i).to_json
   end
 
+  # GET /v3/netsim/<shard-id>?t=<table1>@<id1>&t=<table2>@<id2>&...
```
Comment (islemaster) [inline shared/middleware/net_sim_api.rb]: Correction to this route documentation: `t` => `t[]`

File: shared/middleware/net_sim_api.rb [hunk 3]
```diff
@@ -252,3 +264,19 @@ def has_json_utf8_headers(request)
   end
 
 end
+
+# Convert a query_string of the form "t=table1@1&t=table2@1" into a
```
Comment (islemaster) [inline shared/middleware/net_sim_api.rb]: Also `t` => `t[]`, I think.

File: shared/test/test_net_sim_api.rb [hunk 4]
```diff
@@ -71,6 +71,26 @@ def test_create_read_update_delete
...
 
+  def test_read_multiple_tables
+    create_record({name: 'rec1_1'}, 'table1')
+    create_record({name: 'rec1_2'}, 'table1')
+    create_record({name: 'rec2_1'}, 'table2')
+    create_record({name: 'rec2_2'}, 'table2')
+    create_record({name: 'rec3_1'}, 'table3')
+
+    @net_sim_api.get "/v3/netsim/#{@shard_id}?t[]=table1&t[]=table2@2&t[]=table3@2"
+    assert_equal 200, @net_sim_api.last_response.status
```
Comment (islemaster) [inline shared/test/test_net_sim_api.rb]: This route appears to also respond to `/v3/netsim/#{@shard_id}` without a query string.  Is that a valid request, and should we have a short test to pin that behavior?

**Comment by islemaster** [comment]
Looks good to me after explicitly documenting behavior of the new route with no querystring. :+1:

File: shared/middleware/net_sim_api.rb (hunk 2 above)
Comment (philbogle) [inline shared/middleware/net_sim_api.rb]: Done.

File: shared/middleware/net_sim_api.rb (hunk 3 above)
Comment (philbogle) [inline shared/middleware/net_sim_api.rb]: Done.

**Comment by philbogle** [comment]
Thanks for the review.

**Pull request merged**